### PR TITLE
Add a check before getting the framepayload

### PR DESCRIPTION
**Pull request by oliviamiller**
Ran into a panic when one of the devices sent a packet with no frame payload,  the framePayload will index phyPayload[9:8] and panic. Adding a check to just error and prevent the panic. 

Also adding meta.json

### CODE REVIEW

File: gateway/uplink.go [hunk 1]
```diff
@@ -47,7 +47,12 @@ func (g *Gateway) parseDataUplink(ctx context.Context, phyPayload []byte) (strin
 	// frame port specifies application port - 0 is for MAC commands 1-255 for device messages.
 	fPort := phyPayload[8+foptsLength]
 
-	// device data in the message.
+	// Ensure there is a frame payload in the packet.
+	if int(8+foptsLength+1) >= (len(phyPayload) - 4) {
+		return "", map[string]interface{}{}, errors.New("device sent packet with no data")
```
Comment (JohnN193) [inline gateway/uplink.go:52]: [opt] maybe add device name, though after talking with olivia it sounded like there wasn't much a user could do for this

**Review by JohnN193 — APPROVED** [review]

File: gateway/uplink.go (hunk 1 above)
Comment (oliviamiller) [reply gateway/uplink.go]: added the name, not really anything the user can do to fix this but still may be helpful to know which device is doing this.

File: meta.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        {
+            "api": "rdk:component:sensor",
+            "model": "viam:lorawan:sx1302-gateway"
+         },
+         {
+            "api": "rdk:component:sensor",
+            "model": "viam:lorawan:node"
+         }
+    ],
+    "entrypoint": "mymod"
```
Comment (randhid) [inline meta.json:17]: might want to make this something more standard in the future. Not for this pr.

**Review by randhid — APPROVED** [review]

File: meta.json (hunk 2 above)
Comment (oliviamiller) [reply meta.json:17]: yep will change when I do the CI builds.

**Pull request merged**